lib/layer/styleParser.mjs

```diff
@@ -132,312 +143,395 @@ export default function styleParser(layer) {
-      console.warn(`Layer: ${layer.key}, cannot use both layer.style.hover and layer.style.hovers. Layer.style.hover has been deleted.`);
-      delete layer.style.hover;
-    }
+The layer.style.zIndex legacy configuration is checked.
 
-    // Handle deprecated layer.style.label and layer.style.labels.
-    if (layer.style?.labels && layer.style?.label) {
-      console.warn(`Layer: ${layer.key}, cannot use both layer.style.label and layer.style.labels. Layer.style.label has been deleted.`);
-      delete layer.style.label;
-    }
+The layer.icon_scaling legacy configuartion is checked.
 
-    // Handle layer.style.zIndex deprecation.
-    if (layer.style.zIndex) {
-      console.warn(`Layer: ${layer.key}, layer.style.zIndex has been deprecated. Use layer.zIndex instead.`);
-    }
+@param {layer} layer A json layer object.
+@property {Object} [layer.cluster] Cluster configuration for a point layer.
+@property {Object} [layer.hover] Legacy configuration for style.hover.
+@property {Object} [layer.icon_scaling] Legacy configuration for style.icon_scaling.
+@property {layer-style} layer.style The mapp-layer style configuration.
+@property {feature-style} style.default Default feature style.
+@property {integer} [style.zIndex] Legacy configuration for layer.zIndex.
+*/
+function warnings(layer) {
 
-    if (layer.icon_scaling) {
+  if (!layer.style.default) {
 
-      console.warn(`Layer: ${layer.key}, layer.icon_scaling has been assigned to layer.style.icon_scaling`)
+    console.warn(`Layer: ${layer.key} has no implicit default style. Please add style.default.`)
 
-      layer.style.icon_scaling ??= layer.icon_scaling
-    }
+    // Assign default style for vector layer.
+    // Non cluster layer do not have a default icon style
+    layer.style.default = layer.cluster
+      ? {
+        icon: {
+          type: 'dot'
+        }
+      } : {
+        strokeColor: '#333',
+        fillColor: '#fff9',
+      }
   }
 
-  function parseTheme(theme, layer) {
+  // The default is a feature-style which must not contain a style object property.
+  if (layer.style.default.style) {
 
-    if (typeof theme === 'string') {
+    console.warn(`Layer: ${layer.key} has a style object within the default style configuration.`)
 
-      // Attempt theme lookup in themes from string[key]
-      theme = layer.style.themes?.[theme]
-    }
+    layer.style.default = layer.style.default.style
 
-    if (typeof theme.style === 'object') {
+    delete layer.style.default.style
+  }
 
-      // Assign the default style to the theme.style
-      theme.style = {
-        ...structuredClone(layer.style.default),
-        ...theme.style
-      }
-    }
+  iconObject(layer.style.default)
 
-    if (typeof theme.cat === 'object') {
+  // Handle legacy layer.hover configuration.
+  if (layer.hover) {
+    console.warn(`Layer: ${layer.key}, layer.hover{} should be defined within layer.style{}.`);
+    layer.style.hover = layer.hover;
+    delete layer.hover;
```

Comment:
It appears that the legacy property deletions are happening in the different functions but are not changing the actual layer.

Reply:
In the local scope of the function I can see the deletion happening, but after we kick out of the function the property is persists.